emu/src/io_register_internal_memory/timer_registers.rs

```diff
@@ -0,0 +1,88 @@
+use crate::io_register_internal_memory::io_registers::{IORegister, IORegisterAccessControl};
+
+pub struct TimerRegisters {
+    /// Timer 0 Counter/Reload
+    pub tm0cnt_l: IORegister,
+    /// Timer 0 Control
+    pub tm0cnt_h: IORegister,
+    /// Timer 1 Counter/Reload
+    pub tm1cnt_l: IORegister,
+    /// Timer 1 Control
+    pub tm1cnt_h: IORegister,
+    /// Timer 2 Counter/Reload
+    pub tm2cnt_l: IORegister,
+    /// Timer 2 Control
+    pub tm2cnt_h: IORegister,
+    /// Timer 3 Counter/Reload
+    pub tm3cnt_l: IORegister,
+    /// Timer 3 Control
+    pub tm3cnt_h: IORegister,
+}
+
+impl Default for TimerRegisters {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TimerRegisters {
+    pub const fn new() -> Self {
+        use IORegisterAccessControl::*;
+
+        Self {
+            tm0cnt_l: IORegister::with_access_control(ReadWrite),
+            tm0cnt_h: IORegister::with_access_control(ReadWrite),
+            tm1cnt_l: IORegister::with_access_control(ReadWrite),
+            tm1cnt_h: IORegister::with_access_control(ReadWrite),
+            tm2cnt_l: IORegister::with_access_control(ReadWrite),
+            tm2cnt_h: IORegister::with_access_control(ReadWrite),
+            tm3cnt_l: IORegister::with_access_control(ReadWrite),
+            tm3cnt_h: IORegister::with_access_control(ReadWrite),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use IORegisterAccessControl::*;
+
+    #[test]
+    #[should_panic]
+    fn write_only_timer_register() {
+        let reg = IORegister::with_access_control(Write);
+
+        reg.read();
+    }
+
+    #[test]
+    #[should_panic]
+    fn read_only_timer_register_write() {
+        let mut reg = IORegister::with_access_control(Read);
+
+        reg.write(5);
+    }
+
+    #[test]
+    #[should_panic]
+    fn read_only_timer_register_set_byte() {
+        let mut reg = IORegister::with_access_control(Read);
+
+        reg.set_byte(3, 4);
+    }
+
+    #[test]
+    fn test_timers() {
+        let mut reg = IORegister::with_access_control(ReadWrite);
+
+        assert_eq!(reg.read(), 0);
+
+        reg.write(5);
+
+        assert_eq!(reg.read(), 5);
+
+        reg.set_byte(1, 3);
+
+        assert_eq!(reg.read(), (3 << 8) | 5);
+    }
+}
```

Comment:
We don't need these tests right? They are already in `emu/src/io_register_internal_memory/io_registers.rs`